test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -173,13 +173,20 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 	})
 
 	t.Run("security-agent is running", func(tt *testing.T) {
-		_, err := client.VMClient.ExecuteWithError("pgrep -f security-agent")
-		require.NoError(tt, err, "security-agent should be running")
+		var err error
+		require.Eventually(tt, func() bool {
+			_, err = client.VMClient.ExecuteWithError("pgrep -f security-agent")
+			return err == nil
+		}, 10*time.Second, 500*time.Millisecond, "security-agent should be running ", err)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would be more genereous

```suggestion
		}, 1*time.Minute, 500*time.Millisecond, "security-agent should be running ", err)
```